packages/pluginutils/test/attachScopes.ts

```diff
@@ -70,3 +70,76 @@ test('supports catch without a parameter', (t) => {
     attachScopes(ast, 'scope');
   });
 });
+
+test('supports ForStatement', (t) => {
+  const ast = (parse(
+    `
+    for (let a = 0; a < 10; a++) {
+      console.log(a);
+      let b = 10;
+    }
+  `,
+    { ecmaVersion: 2020, sourceType: 'module' }
+  ) as unknown) as import('estree').Program;
+
+  const scope = attachScopes(ast, 'scope');
+  t.falsy(scope.contains('a'));
+  t.falsy(scope.contains('b'));
+
+  const forLoop = ast.body[0] as import('estree').ForStatement & Record<string, any>;
```

Comment:
How about `as ForStatement & {scope: AttachedScope}`, using the type that attachScopes itself uses?